src/Router.php

```diff
@@ -84,4 +84,23 @@ class Router {
         throw new NotFoundRedirectException();
     }
   }
+
+  public static function getRequestedPage(): string {
+    $page = idx(Utils::getGET(), 'page') ?: idx(Utils::getGET(), 'p');
+    if (!is_string($page)) {
+      $page = 'index';
+    }
+
+    return strval($page);
+  }
+
+  public static function isRequestAjax(): bool {
+    return Utils::getGET()->get('ajax') === 'true';
+  }
+
+  // Check to see if the request is going through the router
+  public static function isRequestRouter(): bool {
+    if (self::getRequestedPage() == "index") return false;
```

Comment:
Yea, the code would be better if it was written as @iangcarroll suggested. But I'm not sure why we're saying all requests that don't have the page `index` go through the router.